app/controllers/admin/lead_provider_users_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Admin::LeadProviderUsersController < ApplicationController
+  before_action :set_lead_provider
+
+  def index
+    @users = User.joins(:lead_provider_profile).where(lead_provider_profiles: { lead_provider: @lead_provider })
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(
+      first_name: params.dig(:user, :first_name),
+      last_name: params.dig(:user, :last_name),
+      email: params.dig(:user, :email),
+    )
```

Comment:
Maybe we can put that in the transaction and use `get_or_create`?

Reply:
Why would we want `get_or_create`, as opposed to just `create!`?

Reply:
Maybe not particularly useful in here, but I think `get_or_create` is the better pattern to use when creating user profiles - because one user can have many profiles. 

It is by coincidence of current business logic that we think lead provider won't have any other profiles.

Although, now that I wrote it, we might want to do something like 'make emails unique' and 'have one type of flow for making a new user with a role, and another with just assigning a role to a user'. Arghh, the rabbit holes.

Reply:
From this page, I would be very surprised if an existing user was given new permissions, as opposed to the creation of a new user

Reply:
Emails are already unique

Reply:
Alright, I think it's fine to keep `create!` here. I do have a feeling we will have to worry about assigning people new roles in an intuitive way (with all the admin DfE users will need to do), but definitely not here and not now.

Reply:
Agreed